packages/builder/src/registry.ts

```diff
@@ -318,4 +317,21 @@ export class OnChainRegistry extends CannonRegistry {
 
     return url === '' ? null : url;
   }
+
+  private async logMultiCallEstimatedGas(datas: any, overrides: Overrides): Promise<void> {
+    try {
+      const estimatedGas = await this.contract.estimateGas.multicall(datas, overrides);
+      console.log(`\nEstimated gas: ${estimatedGas}`);
+      const gasPrice = (overrides.maxFeePerGas as BigNumber) || (await this.provider?.getGasPrice());
```

Comment:
`overrides` can have `gasPrice` property as well which takes precedence over the network default value (obtained by provider.getGasPrice)